base/test/test_utils.cpp

```diff
@@ -110,12 +112,34 @@ bool Test_Utils::saveOrCompare(const char* fileName, const unsigned char* dataTo
 
 	if (boost::filesystem::is_regular_file(fileName))
 	{
+		const std::string strFullFileName = fileName;
+		std::string strFileBaseName = strFullFileName;
+		std::string strExtension;
+		std::string nameOfFile;
+
+		const size_t fileBaseIndex = strFullFileName.find_last_of(".");
+		const size_t fileBaseLocation = strFullFileName.find_last_of("/");
+		if (std::string::npos != fileBaseIndex)
+		{
+			strFileBaseName = strFullFileName.substr(0U, fileBaseIndex);
+			strExtension = strFullFileName.substr(fileBaseIndex);
+		}
+		if (std::string::npos != fileBaseLocation)
+		{
+			 nameOfFile = strFullFileName.substr(fileBaseLocation + 1, fileBaseIndex - fileBaseLocation -1);
+		}
+
 		const uint8_t* dataRead = nullptr;
 		unsigned int dataSize = 0U;
 		BOOST_TEST(readFile(fileName, dataRead, dataSize));
 		BOOST_TEST(dataSize == sizeToSC);
-
 		compareRes = CompareData(dataToSC, dataRead, dataSize, tolerance);
+		if(!compareRes)
+		{
+			boost::filesystem::create_directory("./data/SaveOrCompareFail");
```

Comment:
will create_directory throw an error or silently fail if the directory already exists ?

Reply:
If directory already exists it will fail silently